### DISCUSSION TITLE
Manual failover

### DISCUSSION THREAD

**Discussion by xantinium** [apache/kvrocks#2690] [Q&A] [answered]
Greetings!

Is there a way to make a manual failover in **KVRocks Cluster** with some **redis-cli** command?
https://redis.io/docs/latest/commands/cluster-failover/

The only thing I've founded is a list of supported commands:
https://kvrocks.apache.org/docs/supported-commands#cluster-subcommands
(**CLUSTER FAILOVER** command is not presented)

If this feature is not supported for now, will it be supported in the future releases?

**Comment by git-hulk** [comment, marked as answer]
Hi @xantinium 

You can do this by editing the cluster topology information and sending it to all nodes. But I don't recommend maintaining the cluster in this way. The better way is to use [Kvrocks Controller](https://github.com/apache/kvrocks-controller) to do the failover automatically, but it now depends on the external service(one of zookeeper or etcd), and will support the embedding storage by introducing the raft module.

 
> If this feature is not supported for now, will it be supported in the future releases?

AFAIK, we have no plan to do this for now due to Kvrocks's cluster topology being maintained outside Kvrocks itself.

**Reply by xantinium** [reply]
Thanks!
